internal/handlers/common/update.go

```diff
@@ -635,6 +714,71 @@ func extractValueFromUpdateOperator(op string, update *types.Document) (*types.D
 	return doc, nil
 }
 
+// validateRenameExpression validates $rename input on correctness.
+func validateRenameExpression(update *types.Document) error {
+	if !update.Has("$rename") {
+		return nil
+	}
+
+	updateExpression := must.NotFail(update.Get("$rename"))
+
+	doc, ok := updateExpression.(*types.Document)
+	if !ok {
+		return NewWriteErrorMsg(ErrFailedToParse, "Modifiers operate on fields but we found another type instead")
+	}
+
+	iter := doc.Iterator()
+	keys := map[string]struct{}{}
+
+	for {
+		k, v, err := iter.Next()
+		if err != nil {
+			if errors.Is(err, iterator.ErrIteratorDone) {
+				break
+			}
+
+			return err
```

Comment:
According to our previous agreements, this one should either be a lazy error or should be covered by a test.

Reply:
Thanks! I was trying to find any occurences

Reply:
I think this is the only one in this PR :)